File: redome-front/src/app/shared/dto/logistica.doador.workup.dto.ts
import { BaseEntidade } from "../base.entidade";
import { ArquivoVoucherLogistica } from "../model/arquivo.voucher.logistica";
import { TransporteTerrestre } from "../model/transporte.terrestre";

/**
 * Informação dos detalhes da logística necessária a determinado pedido de logística.
 */
export class LogisticaDoadorWorkupDTO extends BaseEntidade {

	private _transporteTerrestre: TransporteTerrestre[];
	private _aereos: ArquivoVoucherLogistica[];
	private _hospedagens: ArquivoVoucherLogistica[];
	private _observacao: string;

    /**
     * Getter transporteTerrestre
     * @return {TransporteTerrestre[]}
     */
	public get transporteTerrestre(): TransporteTerrestre[] {
		return this._transporteTerrestre;
	}

    /**
     * Getter aereos
     * @return {ArquivoVoucherLogistica[]}
     */
	public get aereos(): ArquivoVoucherLogistica[] {
		return this._aereos;
	}

    /**
     * Getter hospedagens
     * @return {ArquivoVoucherLogistica[]}
     */
	public get hospedagens(): ArquivoVoucherLogistica[] {
		return this._hospedagens;
	}

    /**
     * Getter observacao
     * @return {string}
     */
	public get observacao(): string {
		return this._observacao;
	}

    /**
     * Setter transporteTerrestre
     * @param {TransporteTerrestre[]} value
     */
	public set transporteTerrestre(value: TransporteTerrestre[]) {
		this._transporteTerrestre = value;
	}

    /**
     * Setter aereos
     * @param {ArquivoVoucherLogistica[]} value
     */
	public set aereos(value: ArquivoVoucherLogistica[]) {
		this._aereos = value;
	}

    /**
     * Setter hospedagens
     * @param {ArquivoVoucherLogistica[]} value
     */
	public set hospedagens(value: ArquivoVoucherLogistica[]) {
		this._hospedagens = value;
	}

    /**
     * Setter observacao
     * @param {string} value
     */
	public set observacao(value: string) {
		this._observacao = value;
	}


	public jsonToEntity(res: any) {
		throw new Error("Method not implemented.");
	}

}

File: redome-workup/src/main/java/br/org/cancer/redome/workup/dto/LogisticaDoadorWorkupDTO.java
package br.org.cancer.redome.workup.dto;

import java.io.Serializable;
import java.util.List;

import br.org.cancer.redome.workup.model.ArquivoVoucherLogistica;
import br.org.cancer.redome.workup.model.TransporteTerrestre;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * Informações sobre o detalhe da logística necessária ao doador ou material.
 * 
 * @author Pizão.
 *
 */
@NoArgsConstructor
@Getter
@Builder
public class LogisticaDoadorWorkupDTO implements Serializable {
	
	private static final long serialVersionUID = -3652795976912962282L;

	private List<TransporteTerrestre> transporteTerrestre;
	
	private List<ArquivoVoucherLogistica> aereos;

	private List<ArquivoVoucherLogistica> hospedagens;
		
	private String observacao;

	private Long idPedidoWorkup;
	
	/**
	 * @param transporteTerrestre
	 * @param aereos
	 * @param hospedagens
	 * @param observacao
	 */
	@Builder
	public LogisticaDoadorWorkupDTO(List<TransporteTerrestre> transporteTerrestre, List<ArquivoVoucherLogistica> aereos,
			List<ArquivoVoucherLogistica> hospedagens, String observacao, Long idPedidoWorkup) {
		super();
		this.transporteTerrestre = transporteTerrestre;
		this.aereos = aereos;
		this.hospedagens = hospedagens;
		this.observacao = observacao;
		this.idPedidoWorkup = idPedidoWorkup;
	}
	
}
